test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -1044,6 +1051,16 @@ private static String getPendingClusterStateTasks() {
         return null;
     }
 
+    private static String getHotThreads() {
+        try {
+            Response response = adminClient().performRequest(new Request("GET", "/_nodes/hot_threads"));
```

Comment:
It samples only 3 threads by [default](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-hot-threads.html#:~:text=provide%20information%20for.-,Defaults%20to%203.,-If%20you%20are). We should instruct it to sample all threads, e.g. `threads=9999` as suggested by the API documentation. I also suggest we do not ignore idle threads (`ignore_idle_threads=false`) to make sure we get everything, similar to what we do [here](https://github.com/elastic/elasticsearch/blob/b8d7e99cb9eb6f3644a4269a01bb3ddc6651b073/test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java#L1008).